src/Models/Classes/AdapterEntityCache.ts

```diff
@@ -47,12 +47,8 @@ class AdapterEntityCache<T extends IAdapterData> {
         }
     }
 
-    setEntity(
-        key: string,
-        getEntityData: () => Promise<AdapterResult<T>>
-    ): void {
-        const newCachedEntity = new CachedEntity(getEntityData);
```

Comment:
this seems significant. Why do we change this from the ability to delay populating the data to forcing a resolved value? I thought this was being used already.

Reply:
`setEntity` was not being used anywhere, I revised it that way to use it in the `getEntity`. I can revert this change, but again  setEntity is not called anywhere